src/go/rpk/pkg/cli/cmd/cluster/config/import.go

```diff
@@ -140,9 +152,11 @@ func importConfig(
 			if !all && meta.Visibility == "tunable" {
 				continue
 			}
-			oldValue := oldConfig[k]
-			propertyDeltas = append(propertyDeltas, propertyDelta{k, fmt.Sprintf("%v", oldValue), ""})
-			remove = append(remove, k)
+			oldValue, found := oldConfig[k]
+			if found {
+				propertyDeltas = append(propertyDeltas, propertyDelta{k, fmt.Sprintf("%v", oldValue), ""})
+				remove = append(remove, k)
+			}
```

Comment:
Summary of this block:
* Over _all_ values
* If the value is not in our readback config
* If the value was _previously changed_ i.e., was in our old config
* The value is now removed
This looks correct

The other PR did not have the second conditional check, meaning if the value was _not_ previously changed, we would attempt to remove it (for no reason).